workflow-tracing/src/main/java/com/squareup/workflow1/diagnostic/tracing/TracingWorkflowInterceptor.kt

```diff
@@ -481,15 +484,46 @@ public class TracingWorkflowInterceptor internal constructor(
       state = newState
       output?.let { setOutput(it.value) }
       onWorkflowAction(
-          workflowId = session.sessionId,
-          action = delegate,
-          oldState = oldState,
-          newState = newState,
-          output = output
+        workflowId = session.sessionId,
+        action = delegate,
+        oldState = oldState,
+        newState = newState,
+        output = output
       )
     }
   }
 }
 
+@OptIn(ExperimentalWorkflowApi::class)
+private fun WorkflowIdentifier.toLoggingName(): String {
+  return when (val type = getRealIdentifierType()) {
+    is KClass<*> -> type.toLoggingName()
+    is KType -> type.toLoggingName()
+    else -> toString()
+  }
+}
+
+private fun KType.toLoggingName(): String {
+  if (classifier == null) return toString()
+
+  val classifierName = when (val c = classifier) {
+    is KClass<*> -> c.toLoggingName()
+    else -> toString()
+  }
+
+  val params = arguments.map { projection ->
+    when (val type = projection.type) {
+      is KType -> type.toLoggingName()
+      else -> "*"
+    }
+  }
+
+  return if (params.isEmpty()) classifierName else "$classifierName<${params.joinToString(", ")}>"
+}
```

Comment:
Who needs the reflection library when you can write it yourself? 😅

Reply:
I love extension methods so much. Nothing but syntactic sugar, but they make such a huge difference.